### DISCUSSION TITLE
XML Parser Using Field Name Instead of MSH.1

### DISCUSSION THREAD

**Discussion by dprevatt** [nHapiNET/nHapi#371] [Q&A] [answered]
Is there a way for the XMLParser to use the field name as the element name, for example <FieldSeparator> instead of <MSH.1>?

Thank you!

**Comment by milkshakeuk** [comment]
@dprevatt No, as that isn't part of the HL7v2 XML specifications, but there is a feature request for something similar which is part of the specifications https://github.com/nHapiNET/nHapi/issues/301

This is actively being worked on, no ETA on when it will be ready though.

**Comment by milkshakeuk** [comment, marked as answer]
@dprevatt we have just release version `3.2.0` which supports the `LongName` attribute which might be what you need.
see #301 for additional info.
